zboxcore/sdk/moveworker.go

```diff
@@ -158,15 +162,34 @@ func (req *MoveRequest) ProcessMove() error {
 		go func(blobberIdx int) {
 			defer wg.Done()
 			refEntity, err := req.moveBlobberObject(req.blobbers[blobberIdx], blobberIdx)
-			if err != nil {
-				l.Logger.Error(err.Error())
-				return
+
+			if err == nil {
+				objectTreeRefs[blobberIdx] = refEntity
+				return 
+			}
+			
+			select {
+				case wgErrors <- err:
+				default:
 			}
-			objectTreeRefs[blobberIdx] = refEntity
+			l.Logger.Error(err.Error())
+			
 		}(int(pos))
 	}
 
-	wg.Wait()
+	go func() {
+		wg.Wait()
```

Comment:
we are using defer wg.Done, So it won't be blocked because for every wg.Add there is wg.Done . Can you please explain it, how it can be blocked forever.